### PR TITLE
4.5 Backport Bootloader Fixes for H7 based devices 

### PR DESCRIPTION
**Pull request by bugobliterator**
* Disable second core to avoid spurious writes to RAM at boot
* Fix CubeRed bootloader dual endpoint update method

### CODE REVIEW

**Comment by tpwrules** [comment]
This also has the incorrect description for the required bootloader size (needs to be >= 128K instead of >128K).

**Comment by rmackay9** [comment]
I think we're waiting on this PR to produce a 4.5.5-beta2.  If we don't merge this then I think we have a slight regression on CubeRed.

**Comment by rmackay9** [comment]
I think if we don't have a good reason to update the bootloader then we should leave that change out.

**Comment by tridge** [comment]
we should not change CubeOrange bootloader, its a 743

File: Tools/AP_Bootloader/support.cpp [hunk 1]
```diff
@@ -491,19 +526,27 @@ void port_setbaud(uint32_t baudrate)
   check if flash has any ECC errors and if it does then erase all of
   flash
  */
+#define ECC_CHECK_CHUNK_SIZE (32*sizeof(uint32_t))
 void check_ecc_errors(void)
 {
     __disable_fault_irq();
+    // stm32_flash_corrupt(0x8043200);
```
Comment (tridge) [inline Tools/AP_Bootloader/support.cpp:533]: can we add that fn commented out in flash.c?

**Comment by tridge** [comment]
we've decided for the current beta to not do the ECC changes, limit just to the otg2 and the dual-core fixes

**Comment by bugobliterator** [comment]
> we've decided for the current beta to not do the ECC changes, limit just to the otg2 and the dual-core fixes

I have removed the bootloader changes

**Review by tridge — APPROVED** [review]

**Comment by tridge** [comment]
needs bootloader updates

**Comment by bugobliterator** [comment]
> needs bootloader updates

updated

**Comment by rmackay9** [comment]
OK, we will do a 4.5.5-beta2!

**Comment by rmackay9** [comment]
This is included in 4.5.5-beta2

**Pull request merged**